File: app/src/main/java/com/softaai/designpatterns/java/creational/prototype/Rectangle.java
package com.softaai.designpatterns.java.creational.prototype;



/**
 * Created by Amol Pawar on 01-04-2019.
 * softAai Apps
 */
public class Rectangle extends Shape {

    public Rectangle(){
        type = "Rectangle";
    }

    @Override
    public void draw() {
        System.out.println("Inside Rectangle::draw() method");
    }
}

File: app/src/main/java/com/softaai/designpatterns/kotlin/creational/prototype/Rectangle.kt
package com.softaai.designpatterns.kotlin.creational.prototype


/**
 * Created by Amol Pawar on 01-04-2019.
 * softAai Apps
 */
class Rectangle : Shape() {
    init {
        shapeType = "Rectangle"
    }
    override fun draw() {
        println("Inside Rectangle::draw() method")
    }
}